File: java-old/src/net/nihlen/bf2/objects/Rotation.java
package net.nihlen.bf2.objects;

/**
 * A game server object rotation. Euler angles.
 * Format: yaw/pitch/roll
 * 
 * @author Alex
 */
public class Rotation {

	public final double yaw, pitch, roll;
	
	public Rotation(double yaw, double pitch, double roll) {
		this.yaw = yaw;
		this.pitch = pitch;
		this.roll = roll;
	}
	
	public static Rotation createRotation(String rotationString) {
		String[] parts = rotationString.split("/");
		if (parts.length == 3) {
			double yaw = Double.parseDouble(parts[0]);
			double pitch = Double.parseDouble(parts[1]);
			double roll = Double.parseDouble(parts[2]);
			return new Rotation(yaw, pitch, roll);
		}
		return new Rotation(0, 0, 0);
	}
	
	@Override
	public String toString() {
		return String.format("%s/%s/%s", yaw, pitch, roll);
	}
	
}

File: src/BF2WebAdmin.Server/Entities/Game/Rotation.cs
using System.Globalization;

namespace BF2WebAdmin.Server.Entities.Game
{
    public class Rotation
    {
        public double Yaw { get; set; }
        public double Pitch { get; set; }
        public double Roll { get; set; }

        public Rotation(double yaw, double roll, double pitch)
        {
            Yaw = yaw;
            Roll = roll;
            Pitch = pitch;
        }

        public static implicit operator Rotation(string rotation)
        {
            return Parse(rotation);
        }

        public static Rotation Parse(string rotation)
        {
            var parts = rotation.Split('/');
            return new Rotation(ParseDouble(parts[0]), ParseDouble(parts[1]), ParseDouble(parts[2]));
        }

        private static double ParseDouble(string part)
        {
            return double.Parse(part, NumberStyles.Any, CultureInfo.InvariantCulture);
        }

        public override string ToString()
        {
            return $"{Yaw}/{Roll}/{Pitch}";
        }
    }
}
